test/e2e/shoot/internal/access_serviceaccount.go

```diff
@@ -0,0 +1,175 @@
+// ClusterRoleBindings for easy cleanup.
+var labelsE2ETestStaticServiceAccountToken = map[string]string{"e2e-test": "serviceaccount-static-access"}
+
+// CreateShootClientFromStaticServiceAccountToken creates a ServiceAccount, a corresponding static token secret (issued
+// by kube-controller-manager), and then creates a new shoot client from it.
+// You should call CleanupObjectsFromStaticServiceAccountTokenAccess to clean up the objects created by this function.
+func CreateShootClientFromStaticServiceAccountToken(ctx context.Context, shootClient kubernetes.Interface, name string) (kubernetes.Interface, error) {
+	return createShootClientFromServiceAccount(ctx, shootClient, name, func(serviceAccount *corev1.ServiceAccount) (string, error) {
+		secret := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: serviceAccount.Namespace}}
+		if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, shootClient.Client(), secret, func() error {
+			secret.Labels = utils.MergeStringMaps(secret.Labels, labelsE2ETestStaticServiceAccountToken)
+			secret.Annotations = utils.MergeStringMaps(secret.Annotations, map[string]string{corev1.ServiceAccountNameKey: serviceAccount.Name})
+			secret.Type = corev1.SecretTypeServiceAccountToken
+			return nil
+		}); err != nil {
+			return "", err
+		}
+
+		timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		defer cancel()
+
+		if err := retry.Until(timeoutCtx, 500*time.Millisecond, func(ctx context.Context) (bool, error) {
+			if err := shootClient.Client().Get(ctx, client.ObjectKeyFromObject(secret), secret); err != nil {
+				return retry.SevereError(err)
+			}
+
+			if len(secret.Data[corev1.ServiceAccountTokenKey]) == 0 {
+				return retry.MinorError(fmt.Errorf("token for secret %s not yet populated by kube-controller-manager", client.ObjectKeyFromObject(secret)))
+			}
+
+			return retry.Ok()
+		}); err != nil {
+			return "", nil
+		}
+
+		return string(secret.Data[corev1.ServiceAccountTokenKey]), nil
+	})
+}
+
+// CleanupObjectsFromStaticServiceAccountTokenAccess cleans up all objects in the shoot created by all calls to
+// CreateShootClientFromStaticServiceAccountToken.
+func CleanupObjectsFromStaticServiceAccountTokenAccess(ctx context.Context, shootClient kubernetes.Interface) error {
+	return flow.Parallel(func(ctx context.Context) error {
+		return shootClient.Client().DeleteAllOf(ctx, &corev1.ServiceAccount{}, client.InNamespace(namespaceE2ETestServiceAccountTokenAccess), client.MatchingLabels(labelsE2ETestStaticServiceAccountToken))
+	}, func(ctx context.Context) error {
+		return shootClient.Client().DeleteAllOf(ctx, &corev1.Secret{}, client.InNamespace(namespaceE2ETestServiceAccountTokenAccess), client.MatchingLabels(labelsE2ETestStaticServiceAccountToken))
+	}, func(ctx context.Context) error {
+		return shootClient.Client().DeleteAllOf(ctx, &rbacv1.ClusterRoleBinding{}, client.MatchingLabels(labelsE2ETestStaticServiceAccountToken))
```

Comment:
In `CreateShootClientFromStaticServiceAccountToken`, ServiceAccount and ClusterRoleBinding are labelled with `labelsE2ETestDynamicServiceAccountTokenAccess` 🤔

Reply:
Good catch 😄